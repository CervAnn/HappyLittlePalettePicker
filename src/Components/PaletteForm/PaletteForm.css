.PaletteForm {
  /* display: flex;
  flex-flow: row wrap;
  justify-content: space-evenly; */
}

#palette-img {
  width: 50%;
  height: 100px;
}

#generatePalette-button {
  border: none;
  background-color: transparent;
  color: #D25923;
  font-size: x-large;
  font-weight: 800;
  text-shadow: 
  -1px -1px 0 #FFF, 
  1px -1px 0 #FFF, 
  -1px 1px 0 #FFF,
  1px 1px 0 #FFF;
}

#generatePalette-container {
  display: flex;
  flex-direction: column;
  justify-content: space-around;
  align-items: center;
  height: 150px;
  width: 100%;
  margin: auto;
}

#generatePalette-container:hover {
  border: 2px solid #FFFFFF;
  width: 25%;
}

form {
  display: flex;
  flex-direction: column;
  align-items: center;
  width: 25%;
}

#savePalette-button {
  width: 25%;
}

#addNameSavePalette-container {
  display: flex;
  justify-content: space-evenly;
  align-items: center;
  width: 100%;
  margin: auto;
}