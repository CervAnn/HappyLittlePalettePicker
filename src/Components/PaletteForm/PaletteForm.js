import React, {Component} from 'react'
import './PaletteForm.css'

class PaletteForm extends Component {
  constructor() {
    super()
    this.state = {
      name: '',
      error: ''
    }
  }

  generateNewColors = (e) => {
    e.preventDefault()
    this.props.generateRandomColors()
  }

  handleChange = (e) => {
    this.setState({[e.target.name]: e.target.value});
  };

  handleSubmit = (e) => {
    e.preventDefault();
    const newPalette = {
      id: Date.now(),
      name: this.state.name,
      palettes: this.props.currentPalette
    }
    this.addNewPalette(newPalette);
    this.clearInputs();
  }

  clearInputs = () => {
    this.setState({folderName: ""})
  }


  render() {
  return (
    <section className="PaletteForm">
      <article id="generatePalette-container">
      <img id="palette-img"
      src={require('../../assets/paint-palette.svg')} 
      onClick={this.generateNewColors}
      alt=""/>
      <button id="generatePalette-button"
      onClick={this.generateNewColors}>
        Generate Palette
      </button>
      </article>
      <form id="addPaletteName-form">
        <label>Add Palette Name</label>
        <input
        id="addFolder-input"
        type="text"
        placeholder="Palette Name"
        name="name"
        value={this.state.name}
        onChange={this.handleChange}
        required/>
        <button>Enter Name</button>
      </form>
      <button>SAVE PALETTE</button>
    </section>
  )
}

}

export default PaletteForm;
