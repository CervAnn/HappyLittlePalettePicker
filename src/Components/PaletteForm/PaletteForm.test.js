import React from 'react';
import { shallow } from 'enzyme';
import { configure } from 'enzyme';
import PaletteForm from './PaletteForm'
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('PaletteForm', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<PaletteForm 
    currentPalette={[{color: "#2C7559", isLocked: false}, {color: "57d576", isLocked: false}, {color: "#a35c95", isLocked: false}, {color: "#461d33", isLocked: false}, {color: "413134", isLocked: false}]}
    generateRandomColors={jest.fn()}
    folders={{ folders: [
      {created_at: "2019-10-16", id: 16556, name: "Bold", updated_at: "2019-10-17"}, 
      {created_at: "2019-10-16", id: 16557, name: "Shy", updated_at: "2019-10-17"}, 
      {created_at: "2019-10-16", id: 16558, name: "Animals", updated_at: "2019-10-17"}, 
      {created_at: "2019-10-16", id: 16559, name: "Winding Down", updated_at: "2019-10-17"}, 
      {created_at: "2019-10-16", id: 16560, name: "Monster Mash", updated_at: "2019-10-17"}
    ]}}
    isLoading={false}
    addNewPalette={jest.fn()}
    />);
  })

  it('should initialize state with empty/falsy values', () => {
    expect(wrapper.state('name')).toEqual('')
    expect(wrapper.state('localFolderID')).toEqual('')
    expect(wrapper.state('error')).toEqual('')
    expect(wrapper.state('isSaved')).toEqual(false)
    expect(wrapper.state('viewForms')).toEqual(false)
  })

  it('should call generateNewColors on a button click', () => {
    const mockEvent = { preventDefault: jest.fn() }

    expect(wrapper.state('name')).toEqual('')

    wrapper.instance().generateNewColors = jest.fn()
    wrapper.instance().forceUpdate();
    wrapper.find('button').at(0).simulate('click', mockEvent);

    expect(wrapper.instance().generateNewColors).toHaveBeenCalled();
  })

  it('should call handleChange on a keydown event', () => {

    const mockEvent = { preventDefault: jest.fn() }

    wrapper.find('#download-img').simulate('click', mockEvent);

    const mockEvent2 = () => {}

    wrapper.instance().handleChange = jest.fn();
    wrapper.instance().forceUpdate();
    wrapper.find('input').at(0).simulate('change', mockEvent2);

  expect(wrapper.instance().handleChange).toHaveBeenCalled();
  });
})