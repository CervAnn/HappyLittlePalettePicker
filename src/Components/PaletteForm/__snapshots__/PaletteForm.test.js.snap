// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaletteForm should match the snapshot with all data passed in correctly 1`] = `
<section
  className="PaletteForm"
>
  <div
    id="initial-palette-container"
  >
    <article
      id="generatePalette-container"
    >
      <img
        alt=""
        id="palette-img"
        onClick={[Function]}
        src={
          Object {
            "ReactComponent": Object {
              "$$typeof": Symbol(react.forward_ref),
              "render": [Function],
            },
            "__esModule": true,
            "default": "paint-palette-white.svg",
          }
        }
      />
      <button
        id="generatePalette-button"
        onClick={[Function]}
      >
        Generate New Palette
      </button>
    </article>
    <article
      id="savePalette-container"
    >
      <div
        id="save-palette-buttons-container"
      >
        <img
          alt=""
          id="download-img"
          onClick={[Function]}
          src={
            Object {
              "ReactComponent": Object {
                "$$typeof": Symbol(react.forward_ref),
                "render": [Function],
              },
              "__esModule": true,
              "default": "download.svg",
            }
          }
        />
        <button
          id="savePalette-button"
          onClick={[Function]}
        >
          Save Current Palette
        </button>
      </div>
    </article>
  </div>
</section>
`;
